src/lib.rs

```diff
@@ -454,14 +491,26 @@ impl<'input, Endian> Unit<'input, Endian>
             Some(gimli::AttributeValue::Addr(addr)) => addr,
             _ => 0,
         };
-        let ranges = debug_ranges.ranges(offset, address_size, base_address)
-            .expect("Range offset should be valid");
-        Some(ranges.collect().expect("Should parse ranges"))
+
+        if let Ok(ranges) = debug_ranges.ranges(offset, address_size, base_address) {
+            if let Ok(ranges) = ranges.collect() {
+                return Some(ranges);
```

Comment:
This whole nested `if let` could be rewritten a little more idiomatically like this:

```rust
debug_ranges.ranges(offset, address_size, base_address)
    .and_then(|ranges| ranges.collect())
    .ok()
```